### PR TITLE
Ensure restfulWSClient-3.0 feature works

### PR DESCRIPTION
**Pull request by andymc12**
For now the restfulWSClient will load everything that the server does - eventually we will split this up better.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.internal.restfulWS-3.0.feature [hunk 1]
```diff
@@ -15,7 +15,7 @@ Subsystem-Name: Internal Java RESTful Services 3.0
  com.ibm.websphere.appserver.org.eclipse.microprofile.config-1.4, \
  io.openliberty.jakarta.validation-3.0, \
  com.ibm.websphere.appserver.org.reactivestreams.reactive-streams-1.0, \
- com.ibm.websphere.appserver.mpConfig-1.4, \
+ com.ibm.websphere.appserver.org.eclipse.microprofile.config-1.4, \
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.internal.restfulWS-3.0.feature:18]: You already have that line

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.internal.restfulWS-3.0.feature:18]: Also, that feature depends on cdi-1.2 and tolerates 2.0 ... you're pulling in cdi-3.0

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Updated so that there is actually a difference between the `restfulWS-3.0` and `restfulWSClient-3.0` features - and added a test to verify that a RESTful client is useable with only the `restfulWSClient-3.0` feature. New #build

File: dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            });
+        }
+        for (Object key : System.getProperties().keySet()) {
+            configProperties.put((String) key, System.getProperty((String)key));
+        }
+    }
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources() {
+        new Throwable("ANDY: Config.getConfigSources").printStackTrace();
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigImpl.java:44]: "ANDY"

File: dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigProviderResolverImpl.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                loader.getParent());
+        }
+        return loader.getParent();
+    }
+
+    private final Map<ClassLoader, Config> configs = new WeakHashMap<>();
+
+    @Override
+    public ConfigBuilder getBuilder() {
+        new Throwable("ANDY: ConfigProviderResolverImpl.getBuilder").printStackTrace();
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigProviderResolverImpl.java:57]: ANDY

File: dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigSourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+public class ConfigSourceImpl implements ConfigSource {
+    static final boolean java2SecurityEnabled = System.getSecurityManager() != null;
+    private final Config config;
+    
+    ConfigSourceImpl(Config config) {
+        this.config = config;
+    }
+    @Override
+    public String getName() {
+        new Throwable("ANDY: ConfigSourceImpl.getName").printStackTrace();
```
Comment (WhiteCat22) [inline dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigSourceImpl.java:28]: ANDY

**Review by WhiteCat22 — CHANGES_REQUESTED** [review]

**Review by WhiteCat22 — APPROVED** [review]

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Minor tweak - new #build

**Comment by andymc12** [comment]
Missed the mailImpl-2.0 feature - new #build

**Comment by andymc12** [comment]
#run-libby-bot

**Pull request merged**